From 3b6bc244f1c6df16e7ed40672b3ca50b20daca24 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:26:07 -0400
Subject: [alsa] hda: Add jackpoll_ms to struct azx

Message-id: <140972823573-207-patchset-jkysela@redhat.com>
Patchwork-id: 90907
O-Subject: [RHEL 7.1 PATCH 206/442] ALSA: hda - Add jackpoll_ms to struct azx
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Keeping a pointer to the jackpoll_ms array in the chip will allow
azx_codec_create to be shared between hda_intel and hda_platform
drivers.  Also modify get_jackpoll_ms to make the jackpoll_ms member
optional, this way a platform driver can leave it out if it's not
needed.

Signed-off-by: Dylan Reid <dgreid@chromium.org>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 749ee287fc937d8026fb459b7574d39745c0cbb9)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/hda_intel.c b/sound/pci/hda/hda_intel.c
index 079e5a4..7cb95de 100644
--- a/sound/pci/hda/hda_intel.c
+++ b/sound/pci/hda/hda_intel.c
@@ -1324,8 +1324,13 @@ static void azx_bus_reset(struct hda_bus *bus)
 
 static int get_jackpoll_interval(struct azx *chip)
 {
- int i = jackpoll_ms[chip->dev_index];
+ int i;
  unsigned int j;
+
+ if (!chip->jackpoll_ms)
+  return 0;
+
+ i = chip->jackpoll_ms[chip->dev_index];
  if (i == 0)
   return 0;
  if (i < 50 || i > 60000)
@@ -3172,6 +3177,7 @@ static int azx_create(struct snd_card *card, struct pci_dev *pci,
  chip->driver_type = driver_caps & 0xff;
  check_msi(chip);
  chip->dev_index = dev;
+ chip->jackpoll_ms = jackpoll_ms;
  INIT_WORK(&chip->irq_pending_work, azx_irq_pending_work);
  INIT_LIST_HEAD(&chip->pcm_list);
  INIT_LIST_HEAD(&chip->list);
diff --git a/sound/pci/hda/hda_priv.h b/sound/pci/hda/hda_priv.h
index c231c11..0a56e8e 100644
--- a/sound/pci/hda/hda_priv.h
+++ b/sound/pci/hda/hda_priv.h
@@ -321,6 +321,7 @@ struct azx {
  int capture_streams;
  int capture_index_offset;
  int num_streams;
+ const int *jackpoll_ms; /* per-card jack poll interval */
 
  /* Register interaction. */
  const struct hda_controller_ops *ops;
-- 
1.7.1